File: tests/python/test_calculator.py
from ha_hello.lib.calculator import Calculator


def test_calculator_init():
    calculator = Calculator()
    assert calculator is not None


def test_calculator_add(capsys):
    calculator = Calculator()
    result = calculator.add(1, 2)
    assert result == 3


def test_calculator_subtract(capsys):
    calculator = Calculator()
    result = calculator.subtract(1, 2)
    assert result == -1


def test_calculator_multiply(capsys):
    calculator = Calculator()
    result = calculator.multiply(1, 2)
    assert result == 2


def test_calculator_divide(capsys):
    calculator = Calculator()
    result = calculator.divide(6, 2)
    assert result == 3

File: tests/cpp/test_calculator.cpp
#include "calculator.hpp"
#include "logger.hpp"
#include <gtest/gtest.h>

TEST(CalculatorTest, Add) {
  Logger::info("CalculatorTest, Add");
  Calculator calculator;
  EXPECT_EQ(calculator.add(1, 2), 3);
}

TEST(CalculatorTest, Subtract) {
  Logger::info("CalculatorTest, Subtract");
  Calculator calculator;
  EXPECT_EQ(calculator.subtract(1, 2), -1);
}

TEST(CalculatorTest, Multiply) {
  Logger::info("CalculatorTest, Multiply");
  Calculator calculator;
  EXPECT_EQ(calculator.multiply(1, 2), 2);
}

TEST(CalculatorTest, Divide) {
  Logger::info("CalculatorTest, Divide");
  Calculator calculator;
  EXPECT_EQ(calculator.divide(6, 2), 3);
}
